### PR TITLE
Get rid of Monoid[RamStmt]

### PR DESCRIPTION
**Pull request by bdahse**
Refactor `RamStmt.Seq(List)` to `RamStmt.Seq(Array)` and use `MutList.push!` in the compiler.

### CODE REVIEW

**Comment by bdahse** [comment]
I will post new benchmark results tomorrow.

File: main/src/library/Fixpoint/Compiler.flix [hunk 1]
```diff
@@ -31,16 +31,15 @@ namespace Fixpoint {
...
-                Map.valuesOf |>
-                List.fold;
-            List.fold(evalFacts :: evalStrata :: Nil)
-    }
+    def compile(cs: Datalog[v], strat: Map[PredSym, Int32]): RamStmt[v] with ToString[v] =
+        let doIt = match Datalog(facts, rules) -> {
+            let stmts = MutList.new();
+            Array.foreach(compileRule(stmts), facts);
+            Map.invert(strat) |>
+            Map.foreach(_ -> restrict(rules) >> compileStratum(stmts));
```
Comment (magnus-madsen) [inline main/src/library/Fixpoint/Compiler.flix:39]: I guess I am confused why the argument passed to foreach is not used?

Comment (bdahse) [reply main/src/library/Fixpoint/Compiler.flix:39]: The first argument is the key. The map is from strata to predicates, i.e. `Map[Int32, PredSym]`. We don't care what number a stratum has, only that we compile them in order.

Comment (magnus-madsen) [reply main/src/library/Fixpoint/Compiler.flix:39]: OK, a future comment would be good, but ill merge now.

**Pull request merged**